### ISSUE TITLE
False delivery failure messages

### ISSUE THREAD

**Issue by ttrubep1** [WhisperSystems/Signal-Android#3297]
Since a few days I experience a strange behaviour of TextSecure 2.14.5 on my OnePlus One with Android 5.0.

I'm chatting with someone and the messages I send definitely get delivered: I see the checkmark under the message and my dialog partner reacts/answers on what I have written.

Then, after some time (half a minute to several minutes) the checkmark disappears, I see a red exclamation mark left of the message and below it reads ```Nicht zugestellt``` (German for ```Not delivered```).

When I open the message details it offers to re-send the message, but this only removed the error message and does not bring back the checkmark.

Do you have any idea what's going wrong here?

**Comment by rhodey** [comment]
@ttrubep1 can you try and reproduce this and submit a debug log as soon as you see the red exclamation mark show up?

Settings > Advanced > Submit debug log

then post the URL here

**Comment by ttrubep1** [comment]
@rhodey Here it is: https://gist.github.com/anonymous/ced0e89674127cebd6b2

**Comment by ttrubep1** [comment]
And it just happened again: https://gist.github.com/anonymous/1502c26ce214f6c978d3
Before I forget: Just before it occured now I updated to TextSecure 2.17.0.

**Comment by rhodey** [comment]
thanks for the logs @ttrubep1, the interesting stack traces in the first one are

```
W/PushDecryptJob(30401): org.whispersystems.libaxolotl.DuplicateMessageException: Received message with old counter: 1 , 0
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.getOrCreateMessageKeys(SessionCipher.java:376)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:308)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:265)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:246)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:214)
W/PushDecryptJob(30401): 	at org.whispersystems.textsecure.api.crypto.TextSecureCipher.decrypt(TextSecureCipher.java:114)
W/PushDecryptJob(30401): 	at org.thoughtcrime.securesms.jobs.PushDecryptJob.handleMessage(PushDecryptJob.java:115)
W/PushDecryptJob(30401): 	at org.thoughtcrime.securesms.jobs.PushDecryptJob.onRun(PushDecryptJob.java:95)
W/PushDecryptJob(30401): 	at org.thoughtcrime.securesms.jobs.MasterSecretJob.onRun(MasterSecretJob.java:18)
W/PushDecryptJob(30401): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/PushDecryptJob(30401): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/PushReceivedJob(30401): Received receipt: (XXXXX, 1434136047051)
W/MmsSmsDatabase(30401): Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, mismatched_identities, network_failures, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, mismatched_identities, NULL AS network_failures, 'sms' AS transport_type FROM sms WHERE (thread_id = 11) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, mismatched_identities, network_failures, 'mms' AS transport_type FROM mms WHERE (thread_id = 11) ORDER BY date_received ASC)
W/PushServiceSocket(30401): Push service URL: https://textsecure-service.whispersystems.org
W/PushServiceSocket(30401): Opening URL: https://textsecure-service.whispersystems.org/v1/messages/+**********11/1434136047051
W/PushServiceSocket(30401): Push service URL: https://textsecure-service.whispersystems.org
W/PushServiceSocket(30401): Opening URL: https://textsecure-service.whispersystems.org/v1/messages/+**********11/1434137659273
W/PushDecryptJob(30401): org.whispersystems.libaxolotl.DuplicateMessageException: Received message with old counter: 1 , 0
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.getOrCreateMessageKeys(SessionCipher.java:376)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:308)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:265)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:246)
W/PushDecryptJob(30401): 	at org.whispersystems.libaxolotl.SessionCipher.decrypt(SessionCipher.java:214)
W/PushDecryptJob(30401): 	at org.whispersystems.textsecure.api.crypto.TextSecureCipher.decrypt(TextSecureCipher.java:114)
W/PushDecryptJob(30401): 	at org.thoughtcrime.securesms.jobs.PushDecryptJob.handleMessage(PushDecryptJob.java:115)
W/PushDecryptJob(30401): 	at org.thoughtcrime.securesms.jobs.PushDecryptJob.onRun(PushDecryptJob.java:95)
W/PushDecryptJob(30401): 	at org.thoughtcrime.securesms.jobs.MasterSecretJob.onRun(MasterSecretJob.java:18)
W/PushDecryptJob(30401): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/PushDecryptJob(30401): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
```

the second log is also very interesting, your device re-registered with Google Cloud Messaging **a lot**

```
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
I/Timeline(21004): Timeline: Activity_launch_request id:org.thoughtcrime.securesms time:928734476
W/ResourceType(21004): For resource 0x01030224, entry index(548) is beyond type entryCount(9)
W/ResourceType(21004): For resource 0x01030224, entry index(548) is beyond type entryCount(9)
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
W/JobConsumer(21004): java.io.IOException: SERVICE_NOT_AVAILABLE
W/JobConsumer(21004): 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
W/JobConsumer(21004): 	at org.thoughtcrime.securesms.jobs.GcmRefreshJob.onRun(GcmRefreshJob.java:66)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
W/JobConsumer(21004): 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
W/GcmRefreshJob(21004): GCM registrationId expired, reregistering...
```

I expect that the duplicate message exceptions are the cause of your messages getting marked as failed, but @moxie0 would know better. the GCM exceptions are definitely concerning but I'm not exactly sure how that would affect TextSecure.

**Comment by moxie0** [comment]
Looks like you have a bad gapps install.

**Comment by ttrubep1** [comment]
@moxie0, do you have any idea how to fix this?

I am using a OnePlus One with the most recent stock ROM (version 12.0-YNG1TAS17L based on Android 5.0.2) with the GApps included, i.e. I did not install the GApps on a custom ROM myself. Google Play Services is at version 7.5.74.

I'm rarely using Google's services, so I disabled unused Google apps on my phone (Google+, Hangouts, Play Books/Video/Games/Music/Kiosk, Google Drive, Streetview) but Play Store and Play Services are still active. Can this be related to the problem?

**Comment by agrajaghh** [comment]
Try to enable all google apps and restart the phone

**Comment by Neposu** [comment]
Same problem with New TSVersion. Log after fresh restart. https://gist.github.com/anonymous/d6f4894d50be49deab72

**Comment by itzsaga** [comment]
I have been experiencing the same issue.
https://gist.github.com/a88ef4ad67c07de44eb9

**Comment by riyapenn** [comment]
Related to https://github.com/WhisperSystems/TextSecure/issues/3014?

**Comment by heyimgay** [comment]
Is still still a problem after #4020?

**Comment by Neposu** [comment]
seems to be solved.

**Comment by TimesEnemy** [comment]
Greets.
I realize this post will be mostly useless, but just as a heads up, I too had this issue where I sent a message to a non-TS/Signal user and I received a delivery failure message.  When I clicked the resend button, I would just get the same failure message.  A few days later, I spoke with the person I was trying to message, and they showed me how they had received six or seven duplicates of my message.  It seems I clicked the resend button around six times.  I do not remember how I resolved this.  It could be an update, or migrating to the beta, or a different magnetic field from the earth ... either way, no debug log.  If it happens again, debug log, debug log, debug log.
This was an Android instance of Signal talking to an iPhone messenger app.
